driver/src/orderbook/streamed/orderbook.rs

```diff
@@ -135,17 +136,23 @@ impl TryFrom<&Path> for Orderbook {
 }
 
 impl StableXOrderBookReading for Orderbook {
-    fn get_auction_data(&self, batch_id_to_solve: U256) -> Result<(AccountState, Vec<Order>)> {
-        // TODO: Handle future batch ids for when we want to do optimistic solving.
-        let state = self.create_state()?;
-        // `orderbook_for_batch` takes the index of the auction that is currently collecting orders and returns
-        // the orderbook for the batch index that is currently being solved. `get_auction_data` passed in the
-        // index for the auction that orders should be valid for (the one currently being solved). Thus we need
-        // to increment it.
-        let (account_state, orders) =
-            state.orderbook_for_batch(Batch::Future(batch_id_to_solve.low_u32() + 1))?;
-        let (account_state, orders) = util::normalize_auction_data(account_state, orders);
-        Ok((account_state, orders))
+    fn get_auction_data<'a>(
+        &'a self,
+        batch_id_to_solve: U256,
+    ) -> BoxFuture<'a, Result<(AccountState, Vec<Order>)>> {
+        async move {
+            // TODO: Handle future batch ids for when we want to do optimistic solving.
```

Comment:
Could we create an issue to go along with this TODO!

Reply:
Added a link to #758